pkg/watch/restricted_enqueue_request_for_owner.go

```diff
@@ -1,227 +0,0 @@
-	for req := range reqs {
-		q.Add(req)
-	}
-}
-
-// Generic implements EventHandler.
-func (e *RestrictedEnqueueRequestForOwner) Generic(
-	_ context.Context,
-	evt event.GenericEvent,
-	q workqueue.TypedRateLimitingInterface[ctrl.Request],
-) {
-	reqs := map[reconcile.Request]any{}
-	e.getOwnerReconcileRequest(nil, evt.Object, reqs)
-	for req := range reqs {
-		q.Add(req)
-	}
-}
-
-// parseOwnerTypeGroupKind parses the OwnerType into a Group and Kind and caches the result.  Returns false
-// if the OwnerType could not be parsed using the scheme.
-func (e *RestrictedEnqueueRequestForOwner) parseOwnerTypeGroupKind(scheme *machineryruntime.Scheme) error {
-	// Get the kinds of the type
-	kinds, _, err := scheme.ObjectKinds(e.OwnerType)
-	if err != nil {
-		e.Log.Error(err, "Could not get ObjectKinds for OwnerType", "owner type", fmt.Sprintf("%T", e.OwnerType))
-		return fmt.Errorf("failed to get ObjectKinds: %w", err)
-	}
-	// Expect only 1 kind.  If there is more than one kind this is probably an edge case such as ListOptions.
-	if len(kinds) != 1 {
-		err = fmt.Errorf("%w: expected exactly 1 kind for OwnerType %T, but found %s kinds",
-			ErrNoUniqueKind, e.OwnerType, kinds)
-		e.Log.Error(nil, "expected exactly 1 kind for OwnerType", "owner type",
-			fmt.Sprintf("%T", e.OwnerType), "kinds", kinds)
-		return err
-	}
-	// Cache the Group and Kind for the OwnerType
-	e.groupKind = schema.GroupKind{Group: kinds[0].Group, Kind: kinds[0].Kind}
-	return nil
-}
-
-// getOwnerReconcileRequest looks at object and builds a map of reconcile.Request to reconcile
-// owners of object that match e.OwnerType.
-func (e *RestrictedEnqueueRequestForOwner) getOwnerReconcileRequest(
-	oldIfAny, object client.Object, result map[reconcile.Request]any,
-) {
-	// Iterate through the OwnerReferences looking for a match on Group and Kind against what was requested
-	// by the user
-	refs := e.getOwnersReferences(object)
-	for _, ref := range refs {
-		// Parse the Group out of the OwnerReference to compare it to what was parsed out of the requested OwnerType
-		refGV, err := schema.ParseGroupVersion(ref.APIVersion)
-		if err != nil {
-			e.Log.Error(err, "Could not parse OwnerReference APIVersion",
-				"api version", ref.APIVersion)
-			return
-		}
-		e.getOwnerReconcileRequestFromOwnerReference(oldIfAny, object, result, ref, refGV)
-	}
-}
-
-func (e *RestrictedEnqueueRequestForOwner) getOwnerReconcileRequestFromOwnerReference(
-	oldIfAny, object client.Object,
-	result map[reconcile.Request]any,
-	ref apimetav1.OwnerReference,
-	refGV schema.GroupVersion,
-) {
-	// Compare the OwnerReference Group and Kind against the OwnerType Group and Kind specified by the user.
-	// If the two match, create a Request for the objected referred to by
-	// the OwnerReference.  Use the Name from the OwnerReference and the Namespace from the
-	// object in the event.
-	if ref.Kind != e.groupKind.Kind || refGV.Group != e.groupKind.Group {
```

Comment:
To reviewer:

this condition prevent following logic to enqueue related kyma CR, due to `e.groupKind` is empty

here is some log based on this [commit](https://github.com/kyma-project/lifecycle-manager/pull/1792/commits/2609f056d261942ed675e2f7b6800b143b02a610#diff-d43b6cad816120c3cd2afecb269bcc63171dfa42da998f1e196c6cc344181f56R162)
```
{"level":"INFO","date":"2024-08-23T11:17:04.906351175Z","caller":"watch/restricted_enqueue_request_for_owner.go:162","msg":"RestrictedEnqueueRequestForOwner","context":{"groupKind.Kind":"","groupKind.Group":"","ref.Kind":"Kyma","refGV.Group":"operator.kyma-project.io"}}
```